gradle/lib/dependabot/gradle/update_checker/version_finder.rb

```diff
@@ -160,47 +167,97 @@ def google_version_details
-              namespace = Gradle::FileParser::RepositoriesFinder
-              central = namespace::CENTRAL_REPO_URL
-              raise if repository_url == central
+              raise if central_repo_urls.include?(repository_details["url"])
 
               Nokogiri::XML("")
             end
         end
 
         def repository_urls
-          plugin? ? plugin_repository_urls : dependency_repository_urls
+          plugin? ? plugin_repository_urls : dependency_repository_details
+        end
+
+        def check_response(response, repository_url)
+          return unless [401, 403].include?(response.status)
+          return if @forbidden_urls.include?(repository_url)
+          return if central_repo_urls.include?(repository_url)
+
+          @forbidden_urls << repository_url
+        end
+
+        def repositories
+          return @repositories if @repositories
+
+          details = if plugin?
+                      plugin_repository_details +
+                        credentials_repository_details
+                    else
+                      dependency_repository_details +
+                        credentials_repository_details
+                    end
+
+          @repositories =
+            details.reject do |repo|
+              next if repo["password"]
+
+              # Reject this entry if an identical one with a password exists
+              details.any? { |r| r["url"] == repo["url"] && r["password"] }
+            end
         end
 
-        def dependency_repository_urls
+        def credentials_repository_details
+          credentials.
+            select { |cred| cred["type"] == "maven_repository" }.
+            map do |cred|
+            {
+              "url" => cred.fetch("url").gsub(%r{/+$}, ""),
+              "username" => cred.fetch("username", nil),
+              "password" => cred.fetch("password", nil)
+            }
+          end
+        end
+
+        def dependency_repository_details
           requirement_files =
             dependency.requirements.
             map { |r| r.fetch(:file) }.
             map { |nm| dependency_files.find { |f| f.name == nm } }
 
-          @dependency_repository_urls ||=
+          @dependency_repository_details ||=
             requirement_files.flat_map do |target_file|
               Gradle::FileParser::RepositoriesFinder.new(
                 dependency_files: dependency_files,
                 target_dependency_file: target_file
-              ).repository_urls
+              ).repository_urls.
+                map do |url|
+                  { "url" => url, "username" => nil, "password" => nil }
+                end
             end.uniq
         end
 
-        def plugin_repository_urls
-          [GRADLE_PLUGINS_REPO] + dependency_repository_urls
+        def plugin_repository_details
+          [{
+            "url" => GRADLE_PLUGINS_REPO,
+            "username" => nil,
+            "password" => nil
+          }]
```

Comment:
```suggestion
          }] + dependency_repository_details
```

Reply:
It looks like we used to append the URLs here.  I think we would still want to add them to the end of our new hash.